src/grib_accessor_class_data_ccsds_packing.cc

```diff
@@ -355,32 +360,44 @@ static int pack_double(grib_accessor* a, const double* val, size_t* len)
     binary_scale_factor = grib_get_binary_scale_fact(max, reference_value, bits_per_value, &err);
     divisor             = grib_power(-binary_scale_factor, 2);
 
-    bits8   = (bits_per_value + 7) / 8 * 8;
-    encoded = (unsigned char*)grib_context_buffer_malloc_clear(a->context, bits8 / 8 * n_vals);
+    size_t nbytes = (bits_per_value + 7) / 8;
+    if (nbytes == 3)
+        nbytes = 4;
+
+    encoded = (unsigned char*)grib_context_buffer_malloc_clear(a->context, nbytes * n_vals);
 
     if (!encoded) {
         err = GRIB_OUT_OF_MEMORY;
         goto cleanup;
     }
 
-    buflen = 0;
-    p      = encoded;
-    for (i = 0; i < n_vals; i++) {
-        long blen                  = bits8;
-        unsigned long unsigned_val = (unsigned long)((((val[i] * d) - reference_value) * divisor) + 0.5);
-        while (blen >= 8) {
-            blen -= 8;
-            *p = (unsigned_val >> blen);
-            p++;
-            buflen++;
-        }
+    // ECC-1602: Performance improvement
```

Comment:
Are we assuming a "short" is always 2 bytes and an "int" always 4 bytes?
If so we should have an assert

Reply:
Yes, we have to assume that a short has 2 bytes and int has 4 bytes because this is what libaec assumes. If it's not so, libaec will not work anyway. A check could be useful. Who knows on which systems ecCodes is used?

Reply:
A simple solution would be to use the uint8_t, uint16_t and uint32_t data types.

Reply:
yes that is the correct approach.

Reply:
please use standard headers to bring those types in.